bevy_lint/src/lints/plugin_not_ending_in_plugin.rs

```diff
@@ -0,0 +1,129 @@
+//! Checks for types who implement `Plugin` but whose names does not end in "Plugin".
+//!
+//! This does _not_ check function-style plugins (`fn plugin(app: &mut App)`), only structures with
+//! `Plugin` explicitly implemented with `impl Plugin for T`.
+//!
+//! # Motivation
+//!
+//! Unlike traits like [`Clone`] or [`Debug`], the primary purpose of a type that implements
+//! `Plugin` is to be a Bevy plugin. As such, it is common practice to suffix plugin names with
+//! "Plugin" to signal how they should be used.
```

Comment:
I believe we could expand on the motivation a bit. Currently it explains that this is a common practice, but not really _why_ this is the case (or rather, why it is a good idea to follow it).

Digging into this a bit more will also help to resolve the question whether a `Plugins` suffix should be allowed or not.
If the goal is to clearly distinguish a plugin struct from other structs to ensure that it is used in the correct places, then `Plugins` would probably also accomplish that goal.
If it's more about strict consistency, then perhaps `Plguins` should not be allowed.

Reply:
I've personally always used the "Plugin" suffix for consistency with the ecosystem, but I've always favored consistency in my code.

@alice-i-cecile or @janhohenheim, would you like to pitch in here? Why should we encourage users to use the "Plugin" suffix for all `Plugin`s?